### PR TITLE
fix(web): correct simulator withdrawal calculations

### PR DESCRIPTION
**Pull request by hiroppy**
# Summary

- Calculate taxable withdrawals by grossing up the requested net amount, while capping withdrawals at the available portfolio balance.
- Count only withdrawals actually funded by the portfolio in Monte Carlo return metrics.
- Track cumulative withdrawals, depletion state, and invested principal independently for every sensitivity scenario.
- Apply inflation adjustments to fixed withdrawals at yearly boundaries, matching the simulator UI description.
- Add regression coverage for depletion boundaries and standalone/sensitivity result consistency.

## Linear Issue

- N/A

### Selected Quality Characteristics

| Quality characteristic | Why it is relevant | Acceptance criteria |
| ---------------------- | ------------------ | ------------------- |
| Functional suitability — functional correctness | The change affects balances, withdrawal tax, depletion probability, and security scores. | Funded withdrawals never exceed available assets; sensitivity rows match equivalent standalone runs; inflation increases occur annually. |
| Reliability — fault tolerance | Depleted portfolios and extreme withdrawal requests are high-risk boundary states. | A portfolio that cannot fund a request remains at zero and records only the amount actually delivered without producing invalid probabilities. |
| Maintainability — analysability | Three Monte Carlo path families must use equivalent accounting rules. | Base, contribution-sensitivity, and rate-sensitivity paths use the same withdrawal and failure-accounting semantics, covered by cross-run tests. |

### Test Techniques

| Test technique | Selection rationale | Expected coverage |
| -------------- | ------------------- | ----------------- |
| Boundary value analysis | Portfolio depletion and annual inflation transitions are boundary-sensitive. | Withdrawal larger than balance; first and second inflation-adjusted years. |
| Equivalence partitioning | Base and sensitivity executions should be equivalent for identical effective inputs. | Contribution and withdrawal-rate sensitivity rows match standalone simulations. |
| Decision table testing | Taxable/tax-free, funded/depleted, and amount/rate modes select different calculation branches. | Existing simulator unit suite plus the new branch-specific regressions. |
| State transition testing | Accumulation, withdrawal, and depleted states change how balances and counters evolve. | Actual funded withdrawals and depletion state persist correctly across simulation periods. |

### Primary Test Conditions

- Fixed withdrawals adjusted for inflation remain constant during a year and increase at the next yearly boundary.
- Taxable withdrawals gross up the requested net amount and cannot exceed the current balance.
- An underperforming portfolio with an oversized withdrawal records depletion and principal loss.
- Contribution and rate sensitivity rows produce the same balance, depletion probability, and security score as equivalent standalone runs.
- Sensitivity scenarios compare recovered assets against their own invested principal.

### Out-of-Scope Risks

- Pension start-year bucket semantics were not changed because this PR is limited to the confirmed calculation defects.
- Monte Carlo model assumptions such as return distribution and fixed real contribution semantics were not changed.

## Verification

- [x] Unit tests
- [x] Type checking
- [x] Lint
- [ ] Storybook tests
- [ ] E2E tests
- [ ] Manual verification

### Commands Executed

```text
pnpm exec vitest run --project unit src/components/charts/compound-simulator/calculate-compound.test.ts src/components/charts/compound-simulator/simulate-monte-carlo.test.ts — passed, 115 tests
pnpm --filter @mf-dashboard/web test:unit — passed, 47 files / 648 tests
pnpm --filter @mf-dashboard/web typecheck — passed
pnpm lint — passed
pnpm format:check — passed
pnpm test — simulator-related packages passed, but the run failed in packages/db/src/demo-data.test.ts because 21 demo asset-history rows are later than the current snapshot date; the parallel web task then exited with EPIPE
```

### Checks Not Run

- Storybook tests — no stories or rendered UI changed.
- E2E tests — calculation behavior is covered directly by deterministic unit tests.
- Manual verification — no browser-facing layout or interaction changed.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved inflation-adjusted withdrawal calculations, ensuring annual increases occur at the correct times.
  * Corrected tax handling for withdrawals and accurately recorded the net amount received.
  * Prevented withdrawals from exceeding the available portfolio balance.
  * Improved portfolio depletion and failure calculations when withdrawals are unaffordable or unnecessary.

* **Tests**
  * Added coverage for withdrawal timing, tax adjustments, balance limits, and simulation consistency across varying withdrawal rates and contributions.

### CODE REVIEW

File: apps/web/src/components/charts/compound-simulator/calculate-compound.test.ts [hunk 1]
```diff
@@ -677,10 +677,26 @@ describe("calculateCompound", () => {
...
+        contributionYears: 0,
+        withdrawalStartYear: 2,
+        monthlyWithdrawal: 100_000,
+        withdrawalYears: 2,
+        inflationRate: 10,
+        inflationAdjustedWithdrawal: true,
+      });
+
+      expect(result[3].yearlyWithdrawal).toBe(1_200_000);
+      expect(result[4].yearlyWithdrawal).toBe(1_320_000);
```
[the remark this answers is not in the document]
Comment (hiroppy) [reply apps/web/src/components/charts/compound-simulator/calculate-compound.test.ts:699]: The requested QA approach is already documented in the PR description under “Selected Quality Characteristics,” “Test Techniques,” “Primary Test Conditions,” and “Out-of-Scope Risks.” It explicitly covers functional correctness, annual inflation boundaries, standalone/sensitivity equivalence, depletion boundaries, and the model assumptions left out of scope. No code or PR-body change is needed for this finding.

File: apps/web/src/components/charts/compound-simulator/calculate-compound.ts [hunk 2]
```diff
@@ -106,21 +105,26 @@ export function calculateCompound({
...
           baseWithdrawal = rateBasedMonthlyWithdrawal;
         } else {
-          baseWithdrawal = currentMonthlyWithdrawal;
-          if (inflationAdjustedWithdrawal) {
-            currentMonthlyWithdrawal *= monthlyInflationFactor;
-          }
+          const withdrawalYearIndex = year - withdrawalStartYear - 1;
+          baseWithdrawal = inflationAdjustedWithdrawal
+            ? monthlyWithdrawal * Math.pow(monthlyInflationFactor, withdrawalYearIndex * 12)
+            : monthlyWithdrawal;
```
[the remark this answers is not in the document]
Comment (hiroppy) [reply apps/web/src/components/charts/compound-simulator/calculate-compound.ts:111]: Fixed in 91bcc50. Amount-mode Monte Carlo withdrawals now decline in real value within each year and reset at annual boundaries when inflation adjustment is enabled, matching the deterministic yearly cadence. Added a regression test for equal real annual withdrawal totals. Verified with the full web unit suite (47 files, 649 tests), web typecheck, lint, and format check.

File: apps/web/src/components/charts/compound-simulator/simulate-monte-carlo.ts [hunk 3]
```diff
@@ -108,11 +108,10 @@ export function simulateMonteCarlo({
...
   const isRateMode = annualWithdrawalRate != null && annualWithdrawalRate > 0;
-  // MC operates in real (inflation-adjusted) terms.
-  // - Nominal fixed withdrawal decreases in real terms → deflate each month
-  // - Inflation-adjusted withdrawal stays constant in real terms → no adjustment
-  const monthlyRealWithdrawalFactor =
-    !isRateMode && !inflationAdjustedWithdrawal && ri > 0 ? 1 / Math.pow(1 + ri, 1 / 12) : 1;
+  // MC operates in real (inflation-adjusted) terms. A nominal withdrawal stays
+  // fixed within each year, so its real value declines monthly. An annually
+  // inflation-adjusted withdrawal resets to its starting real value each year.
+  const monthlyRealWithdrawalFactor = !isRateMode && ri > 0 ? 1 / Math.pow(1 + ri, 1 / 12) : 1;
```
[the remark this answers is not in the document]
Comment (hiroppy) [reply apps/web/src/components/charts/compound-simulator/simulate-monte-carlo.ts:114]: Fixed in 499aaba. Pension and other income now use the same within-year real-value factor as the inflation-adjusted expense before net withdrawal is calculated, including sensitivity paths. Added a regression test covering income close to the expense amount. Verified with the full web unit suite (47 files, 650 tests), web typecheck, lint, and format check.

File: apps/web/src/components/charts/compound-simulator/calculate-compound.ts [hunk 4]
```diff
@@ -106,21 +105,26 @@ export function calculateCompound({
...
-        const netWithdrawal = Math.max(baseWithdrawal - income, 0);
-        yearlyWithdrawalTotal += netWithdrawal;
+        const requestedNetWithdrawal = Math.max(baseWithdrawal - income, 0);
         const gainRatio =
           currentTotal > totalPrincipal ? (currentTotal - totalPrincipal) / currentTotal : 0;
-        const taxOnWithdrawal = netWithdrawal * gainRatio * taxRate;
-        const withdrawalRatio = Math.min(netWithdrawal / currentTotal, 1);
+        const effectiveTaxRate = gainRatio * taxRate;
+        const requiredGrossWithdrawal =
+          requestedNetWithdrawal / Math.max(1 - effectiveTaxRate, Number.EPSILON);
```
[the remark this answers is not in the document]
Comment (hiroppy) [reply apps/web/src/components/charts/compound-simulator/calculate-compound.ts:120]: Fixed in 301a710. The displayed effective withdrawal rate now uses the same gross-up formula as the withdrawal engine: rate / (1 - gainRatio × taxRate). Added an all-gain regression case that distinguishes 5.0% from the previous 4.8% result. Verified with the full web unit suite (47 files, 651 tests), web typecheck, lint, and format check.

File: apps/web/src/components/charts/compound-simulator/simulate-monte-carlo.ts [hunk 5]
```diff
@@ -194,6 +202,10 @@ export function simulateMonteCarlo({
       year > withdrawalStartYear && year <= withdrawalStartYear + withdrawalYears;
 
     if (yearlyWithdrawals) yearlyWithdrawals.fill(0);
+    if (isWithdrawing && !isRateMode && inflationAdjustedWithdrawal) {
+      currentMonthlyWithdrawal = monthlyWithdrawal;
+    }
+    let annualRealIncomeFactor = 1;
```
[the remark this answers is not in the document]
Comment (hiroppy) [reply apps/web/src/components/charts/compound-simulator/simulate-monte-carlo.ts]: Fixed in 301a710. Fixed nominal income now carries its annual real-value deflation across withdrawal years instead of resetting to full value each year. The multi-year regression now verifies the second-year income contribution is reduced by annual inflation. Verified with the full web unit suite (47 files, 651 tests), web typecheck, lint, and format check.
